moveit_core/src/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+add_subdirectory(collision_detection_bullet)
```

Comment:
No it's not strictly necessary. My thought was that it was nice to bury some of this code to avoid having to relocate and change the path to many dozens of source files. I could be convinced to move all those `target_sources` call up the main CMakeLists.txt thus eliding them.

My bigger holdup was whether it was worth reorganizing all the tests. Currently I'm keeping them in place but I'm on the fence about whether they should get moved out into their own `tests` directory alongside `include` and `src`. If we do this, then we can delete all these extra CMakeLists.txt in src/. That would be more consistent with how else I've reorganized things but I'd like someone else's input.

Reply:
I was more thinking about using `add_subdirectory(src/...)`. The separate target sources make sense to me, and would even make more sense if we could group all of these targets into fewer but more meaningful namespaces/contexts. Until then, I think it's ok to leave tests where they are for now.

Reply:
> I was more thinking about using add_subdirectory(src/...).

That sounds slightly better to me as well.

> whether they should get moved out into their own tests directory alongside include and src

I do think it makes much more sense to keep the tests in the individual source folders.
moveit_core might become one library now, conceptually, but it's still pretty big and the separate folders help to show what component has how many tests.